integration-tests/tests/src/tests/sync/mongo-db-client.ts

```diff
@@ -93,6 +93,438 @@ describe.skipIf(environment.missingServer, "MongoDB Client", function () {
   });
 
   describe("MongoDBCollection", function () {
+    const insertedId1 = 1;
+    const insertedId2 = 2;
+    const insertedId3 = 3;
+    const insertedText = "Test document";
+    const nonExistentId = 100;
+
+    async function insertThreeDocuments(): Promise<void> {
```

Comment:
I think this could be refactored into a beforeEach hook, it's used in the start of most of the statements so will make this cleaner.

Reply:
👍 Yeah was contemplating that at first. I agree that most tests do use it, but I think there may be enough of the other tests (that don't use it as the very first statement) that it justifies having them where it's needed. Otherwise, some tests would (like you're pointing out in another comment) need to be moved to another describe test suite.

I'll think some more about this and see if having it in a beforeEach could be a better choice. I also like seeing "insertThreeDocuments" in the test so the rest makes more sense as well, e.g. when expecting a length to be `3`.

Reply:
If you added it to the `describe` for the cases that do call `insertThreeDocuments`, it would be fine.

Reply:
How about this? 

```ts
describe("#count", function () {
  describe("after insertions", function () {
       beforeEach(async function (...) => {
        await insertThreeDocuments();
       });

    // Test (B)
    it("returns number of documents using query selector", ...) {
      // ...
    });
  });
  describe("special cases", function () {
            // Test (C)
            it("returns zero when there are no matches", ...) {

              // Never calls the method

              const count = await collection.count({ _id: nonExistentId });
              expect(count).to.equal(0);
            });
  });
});
```
Of course wording can be changed, but this would be quite organised

Reply:
Do you mean having separate `beforeEach`s for each function tested that need to (e.g. `#count`, `#find`)? I may also be misinterpreting the example, but it looks like (for the case of `#count`) the `describe("after insertions", ..)` would only contain 1 `it` test. Not sure if having a `beforeEach` for a suite with 1 test is needed? Again, correct me if my interpretation is off 😛